lib/wallet/http.js

```diff
@@ -767,9 +768,40 @@ class HTTP extends Server {
     this.get('/wallet/:id/tx/history', async (req, res) => {
       const valid = Validator.fromRequest(req);
       const acct = valid.str('account');
-      const txs = await req.wallet.getHistory(acct);
+      const reverse = valid.bool('reverse', false);
+      const limit = valid.u32('limit', 10);
+      const after = valid.brhash('after');
+      const date = valid.str('date');
+      let time = valid.u32('time');
+
+      if (date && !time) {
+        time = util.time(date);
+        enforce(time, 'Invalid date.');
+      }
+
+      enforce(limit <= this.options.maxTxs,
+              `Limit above max of ${this.options.maxTxs}.`);
+
+      let txs = [];
 
-      common.sortTX(txs);
```

Comment:
I think this code block could be made easier to read. Maybe something like:

```javascript
let opts = {
  limit,
  reverse
};

if (after)
  opts.hash = after;
else if (time)
  opts.time = time;

const txs = await req.wallet.listHistoryByTime(acct, opts);
```

I also had the idea of `switch`ing on a bitfield but I think that could just be adding extra complexity that isn't needed

Reply:
There are different functions for after, time and standard queries.

Reply:
Totally, I misread them. Still think that there could be a way to make this easier to read

Reply:
I had it this way so it's more clear to follow. You could separate out the opts with initial values for limit, and reverse. However, I don't really think that saves much or make it necessary more clear.

Reply:
FWIW: It reduces lines by 5.